Storage/Bookmarks/Bookmarks.swift

```diff
@@ -22,6 +22,15 @@ public protocol SyncableBookmarks: class, ResettableSyncStorage, AccountRemovalD
     func applyLocalOverrideCompletionOp(op: LocalOverrideCompletionOp, itemSources: ItemSources) -> Success
 }
 
+public enum BookmarksEditState {
```

Comment:
This might be more accurate if you have `NoUnmergedRemoteChanges` and `UnmergedRemoteChanges` — it's the UI that's deciding that those two should be interpreted as RW/RO.